<launch>
    <arg name="first_agv" default="agv_0" />
    <arg name="second_agv" default="agv_1" />

    <group ns="$(arg first_agv)/map_merge">
        <param name="init_pose_x" value="0.0" />
        <param name="init_pose_y" value="0.0" />
        <param name="init_pose_z" value=" 0.0" />
        <param name="init_pose_yaw" value=" 0.0" />
    </group>

    <group ns="$(arg second_agv)/map_merge">
        <param name="init_pose_x" value="2.0" />
        <param name="init_pose_y" value="-1.0" />
        <param name="init_pose_z" value=" 0.0" />
        <param name="init_pose_yaw" value=" 0.0" />
    </group>
    
    <node pkg="multirobot_map_merge" type="map_merge" respawn="false" name="map_merge">
        <param name="robot_map_topic" value="map" />
        <param name="robot_namespace" value="agv" />
        <param name="merged_map_topic" value="map" />
        <param name="world_frame" value="map" />
        <param name="known_init_poses" value="true" />
        <param name="merging_rate" value="0.5" />
        <param name="discovery_rate" value="0.05" />
        <param name="estimation_rate" value="0.1" />
        <param name="estimation_confidence" value="1.0" />
    </node>
    <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_$(arg first_agv)_tf_broadcaster"
        args="0 0 0 0 0 0 /map /$(arg first_agv)/map" />
    <node pkg="tf2_ros" type="static_transform_publisher"
        name="world_to_$(arg second_agv)_tf_broadcaster"
        args="0 0 0 0 0 0 /map /$(arg second_agv)/map" />
</launch>